File: tagbag/tag_base.py
from pathlib import Path
from typing import List

class TagBase:
    """
    Base class with shared file discovery.
    """

    @staticmethod
    def get_tag_files(path: str | Path) -> List[str]:
        """
        Recursively return tag files under path. First try *.txt. If none are found, try *.npz.
        """
        p = Path(path)
        try:
            txt = [str(f) for f in p.rglob("*.txt")]
            if txt:
                return txt
            npz = [str(f) for f in p.rglob("*.npz")]
            return npz
        except FileNotFoundError:
            return []

File: tagbag/TagBase.cs

namespace NhlStenden.TagBag;

/// <summary>
/// Represents the base class for handling tags.
/// </summary>
public class TagBase
{
    /// <summary>
    /// Retrieves the tag files from the specified directory path.
    /// </summary>
    /// <param name="path">The directory path to search for tag files.</param>
    /// <returns>An array of file names matching the tag file patterns.</returns>
    public static string[] GetTagFiles(string path)
    {
        try
        {
            var fileNames = Directory.GetFiles(path, "*.txt", SearchOption.AllDirectories);
            if (fileNames.Length == 0)
                fileNames = Directory.GetFiles(path, "*.npz", SearchOption.AllDirectories);
            return fileNames;
        }
        catch (DirectoryNotFoundException)
        {
            return [];
        }
    }
}
